docx4j-openxml-objects/src/main/java/org/docx4j/wml/Id.java

```diff
@@ -121,28 +121,28 @@ public void afterUnmarshal(Unmarshaller unmarshaller, Object parent) {
         setParent(parent);
     }
 
-    public boolean equals(Object obj) {
-    	if (obj instanceof Id) {
-	    		return val.equals( ((Id)obj).getVal() ); 
-	    	} else {
-	    		return false;
-	    	}
-	    }
-	   
-	    public int hashCode() {
-	    	
+    public boolean equals(Object obj) {
+    	if (obj instanceof Id) {
+	    		return val.equals( ((Id)obj).getVal() );
+	    	} else {
+	    		return false;
+	    	}
+	    }
+
+	    public int hashCode() {
+
 	    	if (val==null) {
 	    		
-	        	java.math.BigInteger newIdVal = java.math.BigInteger.valueOf(Math.abs(new java.util.Random().nextInt()));
+	        	java.math.BigInteger newIdVal = java.math.BigInteger.valueOf(Math.abs(new java.security.SecureRandom().nextInt()));
 	        	this.setVal( newIdVal );
     			log.warn("Generated Id val " + newIdVal);
 	    		
 	    	}
-	    	
+	    	
 	    	// Natural and good enough...
     		return val.intValue();
-	    }
-        
-    
-    
+	    }
+        
+    
+    

```

Comment:
there must be some invisible characters or something that my IDE is replacing, not worth the hassle imo